### PR TITLE
Schedule

### PR DESCRIPTION
**Pull request by cmochsner**
Schedule and view my rides component and associated service. Added links to navbar. Add tripUri to environment files. Updated environment files for production and dev mode.

### CODE REVIEW

File: src/app/components/schedule/schedule.component.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+       );
+
+       this.id=+sessionStorage.getItem("userid");
+
+       this.serv.getRiderTrips(this.id).subscribe(
+        data =>{ this.riderTrips=data
+      }
+      );
+  }
+  addTrip(t: Trip)
```
Comment (IkenoXamos) [inline src/app/components/schedule/schedule.component.ts]: For readability, please have empty lines between method declarations, and perhaps some js-doc comments explaining what they are doing,

File: src/app/components/schedule/schedule.component.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  {
+    this.serv.getTrips().subscribe(
+      data => this.trips=data
+       );
+
+       this.id=+sessionStorage.getItem("userid");
+
+       this.serv.getRiderTrips(this.id).subscribe(
+        data =>{ this.riderTrips=data
+      }
```
Comment (IkenoXamos) [inline src/app/components/schedule/schedule.component.ts]: These curly brackets aren't necessary. Can be structured similarly to the getTrips() method.

File: src/app/components/view-my-rides/view-my-rides.component.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
+<div class="container centerMe">
+    <div class="card mt-2"> 
+        <table class="table table-hover">
```
Comment (IkenoXamos) [inline src/app/components/view-my-rides/view-my-rides.component.html]: Work on reducing code duplication by either creating a shared component for the table, or just one component for the entire view that can be reused in both places.

File: src/app/components/view-my-rides/view-my-rides.component.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    {
+      console.log("No userId associated with login.")
+    }
+    else
+    {
+    this.serv.getRiderTrips(this.id).subscribe(
+      data => this.trips=data
+       );
+    }
+  }
```
Comment (IkenoXamos) [inline src/app/components/view-my-rides/view-my-rides.component.ts:32]: Please add empty lines between methods for readability.

File: src/app/services/schedule-service/schedule.service.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  providedIn: 'root'
+})
+export class ScheduleService
+{
+  public url: string = environment.tripUri;
+
+  constructor(private http: HttpClient) 
+  { 
+  }
+  getTrips(): Observable<Trip[]>
```
Comment (IkenoXamos) [inline src/app/services/schedule-service/schedule.service.ts]: Please add empty lines between methods for readability.

**Review by IkenoXamos — CHANGES_REQUESTED** [review]
Are any of the changes to all of the 3 environment files necessary for this PR?

**Comment by cmochsner** [comment]
We added a new endpoint (tripUri), which is necessary. Other changes are not neccessary but were made to delete unneeded code. I believe the Google API is now being used on the backend and is no longer needed on the frontend.

File: src/app/components/schedule/schedule.component.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+})
+export class ScheduleComponent implements OnInit 
+{
+  public caption='Trip Schedule';
+  public trips: Trip[];
+  public riderTrips: Trip[]=[];
+  public id: number;
+  
+  constructor(private serv: ScheduleService, private auth: AuthService) 
+  {
```
Comment (IkenoXamos) [inline src/app/components/schedule/schedule.component.ts]: For consistency across the codebase, please move opening curly brackets to the same line as the method declaration.

**Pull request merged**